mantidproject/mantid
# Remove RunSelectorPresenter factory

**Description of work.**
Refactors some of the MVP code around add_runs_presenter, mainly RunSelectionPresenterFactory, which is moved out of the view and condensed into the presenter

**To test:**
- Ensure that the new unit tests run
- Open the ISIS SANS GUI, this will exercise the `RunSelectorPresenterFactory` code as part of the initialisation

Fixes #27226. 

*This does not require release notes* because it is a maintenance issue

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
